contracts/buy/PartyBuy.sol

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.5;
+
+// ============ External Imports: Inherited Contracts ============
+// NOTE: we inherit from OpenZeppelin upgradeable contracts
+// because of the proxy structure used for cheaper deploys
+// (the proxies are NOT actually upgradeable)
+import {
+ReentrancyGuardUpgradeable
+} from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import {
+ERC721HolderUpgradeable
+} from "@openzeppelin/contracts-upgradeable/token/ERC721/utils/ERC721HolderUpgradeable.sol";
+// ============ External Imports: External Contracts & Contract Interfaces ============
+import {IERC721VaultFactory} from "../external/interfaces/IERC721VaultFactory.sol";
+import {ITokenVault} from "../external/interfaces/ITokenVault.sol";
+import {IERC721Metadata} from "@openzeppelin/contracts/token/ERC721/extensions/IERC721Metadata.sol";
+import {IWETH} from "../external/interfaces/IWETH.sol";
+import {Party} from "../Party.sol";
+
+contract PartyBuy is Party {
+    // partyStatus Transitions:
+    //   (1) PartyStatus.ACTIVE on deploy
+    //   (2) PartyStatus.WON after successful buy()
+    //   (3) PartyStatus.LOST after successful fail()
+
+    // ============ Internal Constants ============
+
+    // PartyBuy version 1
+    uint16 public constant VERSION = 1;
+
+    // ============ Public Not-Mutated Storage ============
+
+    // the timestamp at which the Party is no longer active
+    uint256 public expiresAt;
+
+    // ============ Public Mutable Storage ============
+
+    // the maximum price that the party is willing to
+    // spend on the token
+    // NOTE: the party can spend *UP TO* 102.5% of maxPrice in total,
+    // and will not accept more contributions than this max amount
+    uint256 public maxPrice;
+
+    // ============ Events ============
+
+    // emitted when the token is successfully bought
+    event Bought(address triggeredBy, address targetAddress, uint256 ethPrice, uint256 ethFee, uint256 totalContributed);
```

Comment:
nit but `ethPrice` sounds like how much ETH costs rn.  `ethSpent` and `ethFeePaid` feels more natural to me but i recognize this is largely a personal perspective